rib/rib.go

```diff
@@ -1217,6 +1250,69 @@ func (r *RIBHolder) doAddMPLS(label uint32, newRIB *aft.RIB) (bool, error) {
 	return implicit, nil
 }
 
+// DeleteLabelEntry removes the MPLS label entry e from the RIB. It returns a
+// boolean indicating whether the entry has been removed, a copy of the entry
+// that was removed, and an error if the message cannot be parsed. Per the gRIBI
+// specification the payload of the entry is not compared the existing entry
+// before deleting it.
+func (r *RIBHolder) DeleteLabelEntry(e *aftpb.Afts_LabelEntryKey) (bool, *aft.Afts_LabelEntry, error) {
+	if e == nil {
+		return false, nil, errors.New("nil Label entry provided")
+	}
+
+	if r.r == nil {
+		return false, nil, errors.New("invalid RIB structure, nil")
+	}
+
+	if _, ok := e.GetLabel().(*aftpb.Afts_LabelEntryKey_LabelUint64); !ok {
+		return false, nil, fmt.Errorf("unsupported label type %T, only uint64 labels are supported, %v", e, e)
+	}
+
+	lbl := uint32(e.GetLabelUint64())
```

Comment:
why is it ok to cast a uint64 down to a uint32? any chance of truncating the value?

Reply:
Labels can actually only be 20-bits, so we know any valid value here is going to be within that range. For simplification in the generated protobufs (per [this spec](https://github.com/openconfig/ygot/blob/master/docs/yang-to-protobuf-transformations-spec.md)) we made all integer types just use their 64-bit values.